### PR TITLE
SDIT-1038 Manual create activity synchronisation

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesResource.kt [hunk 1]
```diff
@@ -1,18 +1,57 @@
...
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.listeners.EventFeatureSwitch
 
 @RestController
 class ActivitiesResource(
   private val activitiesService: ActivitiesService,
   private val eventFeatureSwitch: EventFeatureSwitch,
+  private val telemetryClient: TelemetryClient,
 ) {
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @PostMapping("/activities/{activityScheduleId}")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesResource.kt:27]: The main change is a new endpoint to emulate receiving an activity created event. 
After problems with synchronisations we fix code/data/config and then end up resyncing - this provides a way to run the sync. again manually.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 2]
```diff
@@ -36,20 +36,24 @@ class ActivitiesService(
   private val objectMapper: ObjectMapper,
 ) : CreateMappingRetryable {
 
-  suspend fun createActivity(event: ScheduleDomainEvent) {
+  suspend fun createActivityEvent(event: ScheduleDomainEvent) {
+    createActivity(event.additionalInformation.activityScheduleId)
+  }
+
+  suspend fun createActivity(activityScheduleId: Long) {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:43]: Now have 2 ways into the synchronisation, by event or called directly from the endpoint.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityResourceIntTest.kt [hunk 3]
```diff
@@ -1,16 +1,113 @@
...
+    @Test
+    fun `access forbidden with wrong role`() {
+      webTestClient.post().uri("/activities/1")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `will synchronise the activity`() = runTest {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityResourceIntTest.kt:59]: A happy path test, pinched from the message handling integration test.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityResourceIntTest.kt [hunk 4]
```diff
@@ -1,16 +1,113 @@
...
+      verify(telemetryClient).trackEvent("activity-create-requested", mapOf("dpsActivityScheduleId" to ACTIVITY_SCHEDULE_ID.toString()), null)
+      verify(telemetryClient).trackEvent(
+        eq("activity-create-success"),
+        check { assertThat(it).containsEntry("dpsActivityScheduleId", ACTIVITY_SCHEDULE_ID.toString()) },
+        isNull(),
+      )
+    }
+
+    @Test
+    fun `will return error if anything fails`() = runTest {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityResourceIntTest.kt:90]: A negative test, again pinched from the message handling integration tests.

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**